okio/src/main/java/okio/AsyncTimeout.java

```diff
@@ -49,15 +52,10 @@
   private static final int TIMEOUT_WRITE_SIZE = 64 * 1024;
 
   /**
-   * The watchdog thread processes a linked list of pending timeouts, sorted in
-   * the order to be triggered. This class synchronizes on AsyncTimeout.class.
-   * This lock guards the queue.
-   *
-   * <p>Head's 'next' points to the first element of the linked list. The first
-   * element is the next node to time out, or null if the queue is empty. The
-   * head is null until the watchdog thread is started.
+   * The watchdog thread processes a queue of pending timeouts, sorted in the order to be triggered.
    */
-  private static AsyncTimeout head;
+  private static final Object watchdogLock = new Object();
+  private static volatile DelayQueue<AsyncTimeout> queue;
```

Comment:
there's an explanation here: http://stackoverflow.com/a/7855774
basically it's possible for the JVM to reorder things if volatile isn't used for the DCL here. the volatile on the queue should still be much cheaper than the static synchronized lock on the whole class monitor lock.